### PR TITLE
Fix rails 6.1 deprecation warning (#47)

### LINKED ISSUE #47: Deprecation warning in Rails 6.1
**Is your feature request related to a problem? Please describe.**
So I am using rails 6.1 and get this deprecation warning. 
```DEPRECATION WARNING: Enumerating ActiveModel::Errors as a hash has been deprecated.
In Rails 6.1, `errors` is an array of Error objects,
therefore it should be accessed by a block with a single block
parameter like this:

person.errors.each do |error|
  attribute = error.attribute
  message = error.message
end

You are passing a block expecting two parameters,
so the old hash behavior is simulated. As this is deprecated,
this will result in an ArgumentError in Rails 6.2.
```

**Describe the solution you'd like**
I guess we have to change [promote_errors](https://github.com/rootstrap/yaaf/blob/master/lib/yaaf/form.rb#L33
) a bit, but as I understand this new behavior will not work in rails < 6.1? I could make a PR, but I don't know how to make it compatible with  rails versions < 6.1.

**Additional context**
https://github.com/rails/rails/pull/32313

### PR DESCRIPTION
**Pull request by Shaglock**
### Summary
Fixes #47. 

### Other Information
I noticed that code coverage dropped a little on my local machine, I guess this is because of version conditions, I have no experience in code coverage, but I guess you need to run against different versions and then [merge the results somehow](https://github.com/simplecov-ruby/simplecov#merging-results)? Or maybe you already know the way to do it, I see that there are some GitHub action and code climate badge (which displays ? for me), anyway, this is probably another topic.
Also, I noticed another deprecation warning when running tests 
```yaaf/lib/yaaf/form.rb:56: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call```. 
This can be easily fixed if you add `**` to options in this method
```ruby
def save_models(options)
  models.map { |model| model.save!(**options) }
end
```
I can make another PR for this warning or update this one.

### CODE REVIEW

**Comment by Shaglock** [comment]
I guess CI fails because of [similar case in reek](https://github.com/troessner/reek/blob/v5.6.0/docs/Duplicate-Method-Call.md#edge-cases)?
I can make reek happy if I create separate methods but this feels a bit too verbose, I don't know.  @santib any thoughts?
```ruby
def promote_errors(model)
  model_errors = model.errors
  if rails_version_less_than_6_1?
    promote_errors_hash(model_errors)
  else
    promote_errors_class(model_errors)
  end
end

def promote_errors_hash(model_errors)
  model_errors.each do |attribute, message|
    errors.add(attribute, message)
  end
end

def promote_errors_class(model_errors)
  model_errors.each do |model_error|
    errors.add(model_error.attribute, model_error.message)
  end
end
```

**Comment by santib** [comment]
@Shaglock this is looking good to me! No need to satisfy Reek in this case, you can add it to the ignore list. Would you mind also adding Rails 6.1 to our CI? It'd be a new file here `spec/gemfiles/` and adding it here `.github/workflows/ci.yml`

File: lib/yaaf/form.rb [hunk 1]
```diff
@@ -30,9 +30,16 @@ def save!(options = {})
 
     attr_accessor :models
 
+    # :reek:DuplicateMethodCall
```
Comment (santib) [inline lib/yaaf/form.rb:33]: can we add the exclusion in `.reek.yml` instead of inline?

**Comment by santib** [comment]
@Shaglock I'm fixing the CI in this PR https://github.com/rootstrap/yaaf/pull/50 . Once I merge my PR, you can rebase this PR and we should be ready to go 💪

**Comment by Shaglock** [comment]
> @Shaglock I'm fixing the CI in this PR #50 . Once I merge my PR, you can rebase this PR and we should be ready to go 💪

I am fixing it too :D I think you should use `git: 'https://github.com/rails/rails.git'` instead of `github:` too

File: lib/yaaf/form.rb (hunk 1 above)
Comment (Shaglock) [reply lib/yaaf/form.rb]: yes, indeed. I thought it was global disable in that file

**Comment by santib** [comment]
> > @Shaglock I'm fixing the CI in this PR #50 . Once I merge my PR, you can rebase this PR and we should be ready to go 💪
> 
> I am fixing it too :D I think you should use `git: 'https://github.com/rails/rails.git'` instead of `github:` too

Yes for older versions of bundler the `git` protocol is insecure, but for newer versions it's fine. Anyways I just merged the other PR in case it helps, but I'm ok changing it for the `https` version if you already did it

**Review by santib — APPROVED** [review]
Nice PR, thank you 😄

File: lib/yaaf/form.rb [hunk 2]
```diff
@@ -60,5 +66,10 @@ def handle_transaction_rollback(exception)
       run_callbacks :rollback
       raise exception
     end
+
+    def rails_version_less_than_6_1?
+      ActiveModel::VERSION::MAJOR < 6 ||
+        ActiveModel::VERSION::MAJOR == 6 && ActiveModel::VERSION::MINOR.zero?
```
Comment (brunvez) [inline lib/yaaf/form.rb:72]: We can use `Gem::Version` to check this if I'm not mistaken

```suggestion
      Gem::Version.new(ActiveModel::VERSION) < Gem::Version.new('6.1')
```

I wonder if `ActiveModel::VERSION` it's not already a `Gem::Version` object

**Review by brunvez — APPROVED** [review]
Nice! Thanks for contributing! Just a minor comment but we can just leave it out

File: lib/yaaf/form.rb (hunk 2 above)
Comment (santib) [reply lib/yaaf/form.rb]: @brunvez I merged it but it broke Rails 4 tests, so I'm gonna revert it

File: lib/yaaf/form.rb [hunk 3]
```diff
@@ -60,5 +66,9 @@ def handle_transaction_rollback(exception)
       run_callbacks :rollback
       raise exception
     end
+
+    def rails_version_less_than_6_1?
+      Gem::Version.new(ActiveModel::VERSION) < Gem::Version.new('6.1')
```
Comment (santib) [inline lib/yaaf/form.rb:71]:
```suggestion
      ActiveModel::VERSION::MAJOR < 6 ||
        ActiveModel::VERSION::MAJOR == 6 && ActiveModel::VERSION::MINOR.zero?
```

**Pull request merged**